src/app/machines/views/MachineDetails/MachineNetwork/NetworkTable/NetworkTableConfirmation/NetworkTableConfirmation.test.tsx

```diff
@@ -225,98 +206,87 @@ describe("NetworkTableConfirmation", () => {
+        />,
+        { store }
       );
-      const confirmation = wrapper.find("ActionConfirm");
-      expect(confirmation.prop("eventName")).toBe("updateInterface");
-      expect(confirmation.prop("confirmLabel")).toBe("Mark as connected");
-      expect(confirmation.prop("statusKey")).toBe("updatingInterface");
-      expect(confirmation.prop("submitAppearance")).toBe("positive");
+      expect(
+        screen.getByText(/Are you sure you want to mark it as connected\?/i)
+      ).toBeInTheDocument();
+      expect(
+        screen.getByRole("button", { name: "Mark as connected" })
+      ).toHaveClass("p-button--positive");
     });
 
     it("can display a mark disconnected confirmation", () => {
       const store = mockStore(state);
-      const wrapper = mount(
-        <Provider store={store}>
-          <MemoryRouter>
-            <CompatRouter>
-              <NetworkTableConfirmation
-                expanded={{
-                  content: ExpandedState.MARK_DISCONNECTED,
-                }}
-                nic={nic}
-                setExpanded={jest.fn()}
-                systemId="abc123"
-              />
-            </CompatRouter>
-          </MemoryRouter>
-        </Provider>
+      renderWithBrowserRouter(
+        <NetworkTableConfirmation
+          expanded={{
+            content: ExpandedState.MARK_DISCONNECTED,
+          }}
+          nic={nic}
+          setExpanded={jest.fn()}
+          systemId="abc123"
+        />,
+        { store }
       );
-      const confirmation = wrapper.find("ActionConfirm");
-      expect(confirmation.prop("eventName")).toBe("updateInterface");
-      expect(confirmation.prop("confirmLabel")).toBe("Mark as disconnected");
-      expect(confirmation.prop("statusKey")).toBe("updatingInterface");
-      expect(confirmation.prop("submitAppearance")).toBe("negative");
+
+      expect(
+        screen.getByText(/Are you sure you want to mark it as disconnected\?/i)
+      ).toBeInTheDocument();
+      expect(
+        screen.getByRole("button", { name: "Mark as disconnected" })
+      ).toHaveClass("p-button--negative");
     });
 
     it("can display a disconnected warning", () => {
       const store = mockStore(state);
-      const wrapper = mount(
-        <Provider store={store}>
-          <MemoryRouter>
-            <CompatRouter>
-              <NetworkTableConfirmation
-                expanded={{
-                  content: ExpandedState.DISCONNECTED_WARNING,
-                }}
-                nic={nic}
-                setExpanded={jest.fn()}
-                systemId="abc123"
-              />
-            </CompatRouter>
-          </MemoryRouter>
-        </Provider>
+      renderWithBrowserRouter(
+        <NetworkTableConfirmation
+          expanded={{
+            content: ExpandedState.DISCONNECTED_WARNING,
+          }}
+          nic={nic}
+          setExpanded={jest.fn()}
+          systemId="abc123"
+        />,
+        { store }
+      );
+
+      expect(
```

Comment:
This is slightly confusing, you're first getting the element by using a partial string match, then verifying the same element again using a longer regex.

You should be able to use just `getByText` along with `.toBeInTheDocument` as it throws an error when not found. In case you had troubles matching the entire string due to white spaces you can just test for the disconnected warning: `"This interface is disconnected, it cannot be configured unless a cable is connected"`

Reply:
I did it this way since looking for it by text wouldn't work, since the message is broken up by multiple elements:

![image](https://github.com/canonical/maas-ui/assets/35104482/a4283f39-e711-4659-a268-7447b17ceea7)

I tried using the string you suggested there, but it still results in this same error

Reply:
You should be able to workaround this using a custom text matcher:

```tsx
const getByTextContent = (text: string | RegExp) => {
  return screen.getByText((_, element) => {
    const hasText = (element: Element | null) => {
      if (element) {
        if (text instanceof RegExp && element.textContent) {
          return text.test(element.textContent);
        } else {
          return element.textContent === text;
        }
      } else {
        return false;
      }
    };
    const elementHasText = hasText(element);
    const childrenDontHaveText = Array.from(
      element?.children || []
    ).every((child) => !hasText(child));
    return elementHasText && childrenDontHaveText;
  });
};
```

@Jay-Topher is already using this in another project.